stout/flags.cc

```diff
@@ -99,8 +238,59 @@ void Parser::Parse(int* argc, const char*** argv) {
       break;
     }
 
-    // Skip anything that doesn't look like a flag.
+    // Skip anything that doesn't look like a flag, subcommand or positional
+    // argument.
     if (arg.find("--") != 0) {
+      // It could be probably subcommand.
+      const auto nested_parser_iter = std::find_if(
+          nested_parsers_.begin(),
+          nested_parsers_.end(),
+          [&arg](const auto& parser) {
+            return parser.second.count(arg) > 0;
+          });
+
+      // It's a subcommand.
+      if (nested_parser_iter != nested_parsers_.end()) {
+        // Keep the number of arguments remained before we call 'nested'
+        // Parse() function including subcommand name.
+        int args_remained = *argc - i;
+
+        CHECK_GE(args_remained, 0);
+
+        // Keep subcommand name. Using this pointer we can easily call
+        // Parse(&argc, &argv) function for nested parser.
+        const char** arg_to_parse = &((*argv)[i]);
+
+        // Current argument is a subcommand. So now we can set
+        // 'google::protobuf::Message*' pointer for nested parser
+        // in order to parse flags followed after this subcommand.
+        nested_parser_iter->first->message_for_parsing_ =
+            GetMessageForSubcommand(arg);
+
+        CHECK(nested_parser_iter->first->message_for_parsing_);
+
+        // Parsing arguments for 'nested' parser.
+        nested_parser_iter->first->Parse(&args_remained, &arg_to_parse);
+
+        // If after 'nested' parsing of arguments there were no any arguments
+        // after '--' just break.
+        if (args_remained == 1) {
+          break;
+        } else {
+          // Start at '1' to skip arg_to_parse[0] which is the name
+          // of a subcommand.
+          for (int k = 1; k < args_remained; ++k) {
+            args.push_back(arg_to_parse[k]);
+          }
+          break;
+        }
+      } else {
+        // It might be a positional argument or an unknown argument.
```

Comment:
In the comment you speaking about both unknown and positional arg, but in code you are sure that it is unknown one

Reply:
Yep, cause logic about parsing pos arguments will be added here in the future PR.